implementation/src/main/java/io/smallrye/mutiny/operators/multi/builders/BaseMultiEmitter.java

```diff
@@ -135,4 +157,22 @@ public MultiEmitter<T> onTermination(Runnable onTermination) {
     public MultiEmitter<T> serialize() {
         return new SerializedMultiEmitter<>(this);
     }
+
+    @Override
+    public MultiEmitter<T> onRequest(LongConsumer consumer) {
+        ParameterValidation.nonNull(consumer, "consumer");
+        if (!disposed.get()) {
```

Comment:
We could potentially have a race condition here, as the disposed may be set just after this call.

We may need a few race tests

Reply:
This can happen with a concurrent cancellation, yet since a cancellation is eventual that should be fine.

The other case is a termination (error, completion), but in this case the request itself will have no effect. And we can't control what the callback does, but if it tries to emit further items it won't do anything anyway.